plugins/fabric8-multi-tenant-manager/src/main/java/com/redhat/che/multitenant/TenantDataProvider.java

```diff
@@ -179,6 +118,72 @@ private String getResponseBody(final String endpoint, final String keycloakToken
     return response.asString();
   }
 
+  private class TenantDataLoader extends CacheLoader<CacheKey, UserCheTenantData> {
+
+    @Override
+    public UserCheTenantData load(CacheKey cacheKey) throws InfrastructureException {
+      if (standalone) {
+        String namespaceType = cacheKey.namespaceType;
+        String namespaceToUse;
+        switch (namespaceType) {
+          case "che":
+            namespaceToUse = cheNamespace;
+            break;
+          default:
+            namespaceToUse = "myproject";
+        }
+        return new UserCheTenantData(
+            namespaceToUse, "https://kubernetes.default.svc", "dummy.prefix.unused", false);
+      }
+
+      String responseBody;
+      try {
+        responseBody = getResponseBody(fabric8UserServiceEndpoint, cacheKey.keycloakToken);
+      } catch (ServerException
+          | UnauthorizedException
+          | ForbiddenException
+          | NotFoundException
+          | ConflictException
+          | BadRequestException
+          | IOException e) {
+        throw new InfrastructureException("Exception during the user tenant data retrieval", e);
+      }
+      try {
+        final JsonArray namespaces =
```

Comment:
It is legacy code but still this code is very complex for such a simple thing as getting fields of an object from a response. You could drastically improve this code by refactoring parsing. You could create an object with only fields that we need here and parse response as a single operation instead of traversing it manually. You could use Google GSON used here or switch to Jackson - both libraries are heavily used in upstream Che.

Reply:
Mostly I was avoiding this because I hate dealing with json in Java ;)